File: game/data/plugins/skill/fishing/fish.rb

# The hash of names to fish.
CATCHABLE_FISH = {}

# A fish that can be caught.
class Fish
  attr_reader :id, :level, :experience, :name

  # Creates the Fish.
  def initialize(id, level, experience)
    @id = id
    @level = level
    @experience = experience

    @name = name_of(:item, id)
  end

end

# Appends a Fish to the hash.
def append_fish(name, fish)
  	CATCHABLE_FISH[name] = fish
end

append_fish(:shrimp,    Fish.new(317, 1,  10))
append_fish(:sardine,   Fish.new(327, 5,  20))
append_fish(:herring,   Fish.new(345, 10, 30))
append_fish(:anchovy,   Fish.new(321, 15, 40))
append_fish(:mackerel,  Fish.new(353, 16, 20))
append_fish(:trout,     Fish.new(335, 20, 50))
append_fish(:cod,       Fish.new(341, 23, 45))
append_fish(:pike,      Fish.new(349, 25, 60))
append_fish(:salmon,    Fish.new(331, 30, 70))
append_fish(:tuna,      Fish.new(359, 35, 80))
append_fish(:lobster,   Fish.new(377, 40, 90))
append_fish(:bass,      Fish.new(363, 46, 100))
append_fish(:swordfish, Fish.new(371, 50, 100))
append_fish(:shark,     Fish.new(383, 76, 110))

File: game/plugin/skills/fishing/src/org/apollo/game/plugin/skills/fishing/Fish.kt
package org.apollo.game.plugin.skills.fishing

import org.apollo.game.plugin.api.Definitions

/**
 * A fish that can be gathered using the fishing skill.
 */
enum class Fish(val id: Int, val level: Int, val experience: Double, catchSuffix: String? = null) {
    SHRIMPS(id = 317, level = 1, experience = 10.0, catchSuffix = "some shrimp."),
    SARDINE(id = 327, level = 5, experience = 20.0),
    MACKEREL(id = 353, level = 16, experience = 20.0),
    HERRING(id = 345, level = 10, experience = 30.0),
    ANCHOVIES(id = 321, level = 15, experience = 40.0, catchSuffix = "some anchovies."),
    TROUT(id = 335, level = 20, experience = 50.0),
    COD(id = 341, level = 23, experience = 45.0),
    PIKE(id = 349, level = 25, experience = 60.0),
    SALMON(id = 331, level = 30, experience = 70.0),
    TUNA(id = 359, level = 35, experience = 80.0),
    LOBSTER(id = 377, level = 40, experience = 90.0),
    BASS(id = 363, level = 46, experience = 100.0),
    SWORDFISH(id = 371, level = 50, experience = 100.0),
    SHARK(id = 383, level = 76, experience = 110.0, catchSuffix = "a shark!");

    /**
     * The name of this fish, formatted so it can be inserted into a message.
     */
    val catchMessage by lazy { "You catch ${catchSuffix ?: "a $catchName."}" }

    private val catchName by lazy { Definitions.item(id).name.toLowerCase().removePrefix("raw ") }
}
